src/index.ts

```diff
@@ -423,7 +423,12 @@ const initialize = (apiKey: string, target: Target, options?: Options): Result =
       logDebug('Flag variation has changed for ', evaluation.identifier)
       storage[evaluation.flag] = value
       evaluations[evaluation.flag] = { ...evaluation, value }
-      sendEvent(evaluation)
+
+      // We want to notify for streaming events only.
+      // Polling emits its own events, so don't want to duplicate these.
+      if (configurations.streamEnabled && !poller.isPolling()) {
+        sendEvent(evaluation)
+      }
```

Comment:
So the Flutter SDK emits an 'EVALUATION_POLLING' event to its users, which is a separate event from changed. In order to keep backwards compatibility, I had to do the same thing in the web plugin. As you've alluded to, the CHANGED event was fired for a few different reasons. 
I can either break backwards compatibility with Flutter, or introduce a new event along with the new polling feature. I get where you're coming from with users having to write new code to get change events from polling. Maybe there's a clever way to make it emit for polling by default, but opt out of required (i.e. for Flutter).

Reply:
You could keep both here and ensure there is no mention in the docs about the `POLLING_CHANGED` event. Kinda like a secret API. Also, looking at the code that emits the `POLLING_CHANGED`, it doesn't actually detect if anything changed at all. If no flags change between the initial load and any of the polls, it will still continue to fire the event each time.

Reply:
I didn't appreciate it would change how the user subscribes for the events.  I'd agree it makes sense that they only subscribe for a change event, and receive an event regardless of the mechanism.

Reply:
The secret API makes sense. 
The behaviour you mentioned for change detection - that's specific to the JS SDK. Android emits a the event which triggers Flutters 'EVALUATION_POLLING' anytime the cache is reloaded - which includes on polling intervals. So again, this comes down to keeping behaviour between the plugins identical, until we deem that a major revision is required in Flutter.

Reply:
I'll get this changed tomorrow up tomorrow. I appreciate the input. I know introducing less than optimal UX for a single user isn't a good enough reason, so I'll go with the secret event.

Reply:
Technically speaking, when there are no changes, the cache isn't reloaded. What the JS SDK does is discard anything that's not changed. What could be a better fit might be to add a new `allVariations` function to be able to grab everything at once, then fire a `POLLING_SUCCESS` event. You can then listen for that in the Flutter Web SDK and then grab all evaluations. The downside to that would be that metrics would be fired each time, unless you disabled the metrics while that call is being served.

Something like...

```typescript
const allVariations = ():Record<string, VariationValue> => {
  stopMetricsCollector()
  const allEvaluations = { ...storage }
  startMetricsCollector()

  return allEvaluations
}
```

Reply:
Ok, thank you - this solution sounds good, especially `POLLING_SUCCESS` which is intuitive. 

Maybe adding a simple boolean flag argument to `allVariations`  to call `startMetricsCollector()` would be good enough.

Reply:
It's possible that I can just use the original event structure as you and Dave are alluding to, and have the differentiation in Flutter Web come from whether streaming is enabled or not. If enabled, Flutter web will emit the streaming related event to users. If disabled, it will emit polling.
Note, that streaming and polling for Flutter Web is such that polling can't be disabled, to align with Android and iOS.